it/src/test/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManagerTest.java

```diff
@@ -0,0 +1,312 @@
+  private final String region = "us-east1";
+  private DefaultSpannerResourceManager testManager;
+
+  @Captor private ArgumentCaptor<Iterable<Mutation>> writeMutationCaptor;
+  @Captor private ArgumentCaptor<Iterable<String>> statementCaptor;
+  @Captor private ArgumentCaptor<String> instanceIdCaptor;
+  @Captor private ArgumentCaptor<String> databaseIdCaptor;
+
+  @Before
+  public void setUp() {
+    testManager = new DefaultSpannerResourceManager(spanner, testId, projectId, region);
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerCreateInstanceFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getInstanceAdminClient().createInstance(any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerCreateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getDatabaseAdminClient().createDatabase(any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_throwsExceptionWhenSpannerUpdateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    when(spanner.getDatabaseAdminClient().updateDatabaseDdl(any(), any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTable_worksWhenSpannerDoesntThrowAnyError() {
+    //   arrange
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act
+    testManager.createTable(statement);
+
+    // assert
+    verify(spanner.getDatabaseAdminClient())
```

Comment:
Use a captor and then use Truth to validate the captor's results. This:

1. Is more consistent with other tests (see: https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/it/src/test/java/com/google/cloud/teleport/it/dataflow/DataflowOperatorTest.java)
2. Will create more consistent error messages to other assertions.
3. Will make this a lot easier to read. `updateDatabaseDdl(instanceCaptor.capture(), ...)` is way more clear than what is here. (Honestly, it took me a bit for me to figure out what these were, especially the one for the instance id.)